src/js/containers/explorer/detail/table/ExplorerTableContainer.jsx

```diff
@@ -0,0 +1,259 @@
+                nextEllipses = '';
+                lastButton = '';
+            }
+        }
+
+        // calculate start and end item indexes
+        const startIndex = (currentPage - 1) * pageSize;
+        const endIndex = Math.min(startIndex + (pageSize - 1), (totalItems - 1));
+
+        // create an array of pages to repeat in the pager control
+        const pages = range(startPage, endPage + 1);
+
+        // return object with all pager properties
+        return {
+            totalItems,
+            currentPage,
+            pageSize,
+            totalPages,
+            startPage,
+            endPage,
+            startIndex,
+            endIndex,
+            pages,
+            prevEllipses,
+            nextEllipses,
+            firstButton,
+            lastButton
+        };
+    }
+
+    setPageOfItems() {
+        const page = this.props.pageNumber;
+        const results = this.state.results;
+
+        // Get new pager object for specified page
+        const pager = this.getPager(results.length, page, 20);
+
+        // Get new page of items from results
+        const pageOfItems = results.slice(pager.startIndex, pager.endIndex + 1);
+
+        // update state
+        this.setState({
+            pager,
+            pageOfItems
+        });
+    }
+
+    parseResults(data) {
+        const results = [];
+
+        data.forEach((item) => {
+            // Format obligated amount
+            const formattedCurrency =
+                MoneyFormatter.formatMoneyWithPrecision(item.amount, 0);
+
+            const percentageValue = (item.amount / this.props.total);
+
+            // Convert from decimal value to percentage and round to 2 decimal places
+            const formattedPercentage = (percentageValue * 100).toFixed(2);
+
+            let percent = `${formattedPercentage}%`;
+            if (percent === '0.00%') {
+                percent = 'Less than 0.01%';
+            }
+
+            const result = {
+                name: item.name,
+                id: item.id,
+                obligated_amount: item.amount,
+                percent_of_total: percentageValue,
+                display: {
+                    name: item.name,
+                    obligated_amount: formattedCurrency,
+                    percent_of_total: percent
+                }
+            };
+            results.push(result);
+        });
+
+        this.setState({
+            results
+        }, () => {
+            this.setPageOfItems();
+        });
+    }
+
+    showColumns() {
+        const columns = [];
+        const sortOrder = ExplorerTableFields.defaultSortDirection;
+
+        ExplorerTableFields.order.forEach((col) => {
+            const displayName = ExplorerTableFields[col];
+
+            const column = {
+                columnName: col,
+                displayName,
+                defaultDirection: sortOrder[col]
+            };
+            columns.push(column);
+        });
+
+        this.setState({
```

Comment:
It looks like multiple chains of `setState` are being called via these callbacks. Every time state is updated, the component and all its child components re-render (or at least diff). If we need all these operations (showColumns, parseData, setPages) to be done before a meaningful render can occur, it is better to perform all these operations and then do a single setState to reduce unusable render passes.